DSharpPlus.UnifiedCommands/Message/Internals/Exceptions/ConversionFailedException.cs

```diff
@@ -0,0 +1,17 @@
+namespace DSharpPlus.UnifiedCommands.Message.Internals.Exceptions;
+
+public class ConversionFailedException : Exception
+{
+    public InvalidMessageConversionType Type { get; private set; }
+    public string Value { get; private set; }
+    public bool IsPositionalArgument { get; private set; }
+    public string Name { get; private set; }
+
+    public ConversionFailedException(string value, InvalidMessageConversionType type, bool isPositionalArgument, string name)
```

Comment:
exception constructors should always contain a `string message` and send it to the base constructor, potentially even an inner exception (useful in this case, should we want to allow users to throw their own exceptions in argument converters)